### ISSUE TITLE
Deleted observations don't have a timestamp

### ISSUE THREAD

**Issue by noffle** [digidem/mapeo-core#91]
This could be useful to have, e.g. for reverting (#90).

**Comment by gmaclennan** [comment]
I think it's useful / necessary to have also for "de-forking" a case where one head is a deletion and the other is not. Do you know how this is resolved currently?

**Comment by noffle** [comment]
> I think it's useful / necessary to have also for "de-forking" a case
> where one head is a deletion and the other is not. Do you know how
> this is resolved currently?

Yes, the sort function in osm-p2p-defork prioritizes rows with
timestamps over rows without. I think this is Good Behaviour: I imagine
users would prefer to re-delete than restore (which we have no UI for
yet).

**Comment by gmaclennan** [comment]
Ah! This is good to know. Aliya has mentioned an issue with deletions not working before but I have not been able to reproduce. So if I understand right, if there is a forked node and it is deleted within iD Editor, what the user will see after saving the change is the node re-appear (because it is the forked node appearing)? I think this is potentially confusing because it will seem like delete does not work until you delete each of the forks (whose number the user does not know) so you need to keep deleting the same node and saving until it goes away.

**Comment by noffle** [comment]
> So if I understand right, if there is a forked node and it is deleted
> within iD Editor, what the user will see after saving the change is
> the node re-appear (because it is the forked node appearing)?

No. Whenever a user makes a local edit, they are creating a new, single
head. This conflict would only ever happen after syncing: a user who
made a deletion and then synced with someone who has a forked edit,
would see their deletion vanish.

**Comment by gmaclennan** [comment]
> No. Whenever a user makes a local edit, they are creating a new, single head.

I think local _edits_ (as opposed to deletes) do not merge (edits from both iD and MapFilter pass `opts.links` so that the edit is not a merge operation). Observation deletes [_do_ merge heads](https://github.com/digidem/mapeo-core/blob/master/index.js#L148), but deletes from iD Editor [do not merge](https://github.com/digidem/osm-p2p-server/blob/master/api/put_changes.js#L83) e.g. if there are two heads and the user deletes in iD, there will still be two heads, the deleted head and the other, still undeleted, head.

As far as I understand it, if there are two heads of a node in iD Editor, and the user deletes, then after save the other (non-deleted) head will appear. If we are unsure we can test.

(I think this behaviour has kind of evolved as we go, and we've never made a clear decision about how things should work, and I'm just trying to get clear on what the current behaviour is, then we can figure out what it should be.)

**Comment by noffle** [comment]
Ah yes, that sounds right! Given that, adding a timestamp should fix the problem.
